spark/src/main/scala/geotrellis/spark/io/kryo/GeometrySerializer.scala

```diff
@@ -0,0 +1,70 @@
+package geotrellis.spark.io.kryo
+
+import geotrellis.vector.Geometry
+import geotrellis.vector.io.wkb.WKB
+
+import com.esotericsoftware.kryo.{Kryo, Serializer}
+import com.esotericsoftware.kryo.io.{Input, Output}
+
+import java.io.ByteArrayOutputStream
+import java.util.zip.{Deflater, Inflater}
+
+class GeometrySerializer[G <: Geometry] extends Serializer[G] {
+  override def write(kryo: Kryo, output: Output, geom: G): Unit = {
+    val wkb = WKB.write(geom)
+
+    val (data, compressed) =
+      if (wkb.length > 1400) {
+        // Only bother deflating data larger than about 1 packet worth
+        val deflater = new Deflater
+        deflater.setInput(wkb)
+        deflater.finish
+        val baos = new ByteArrayOutputStream
+        val buffer = Array.ofDim[Byte](4096)
```

Comment:
Btw, we actually always can guess max possible compressed data length: initial data size + 10 header bytes (for compression level 0).

Reply:
I don't think there's much value in exposing a default compression level option to the user.  This stuff should all be transparently more awesome to the user, and the default level balances speed and compression level.  I think the only argument to be made here is do we just put the dial to 11 and max out the compression level under the assumption that network transmission is so slow that the time savings for packet transmission more than make up for the time spent compressing to the highest degree?